### DISCUSSION TITLE
Livewire with Lucid Architecture

### DISCUSSION THREAD

**Discussion by undjike** [lucidarch/lucid#8] [Q&A] [answered]
I don't really know if it's the right place for this question, but, how do I use Livewire with Lucid Architecture ?

**Comment by Mulkave** [comment]
Could you please provide more details on what exactly you would like to know? Since Lucid is simply to structure and run the code.

I haven't had the chance to try Livewire myself but i intend to do so by the beginning of Dec. If you do try it please elaborate with more detailed questions here to be able to discuss further.

**Comment by undjike** [comment]
Okay, I see. All about Livewire is [here](https://laravel-livewire.com/docs/2.x/quickstart). It comes directly with fresh Laravel 8.x projects (so does Jetstream and Fortify). 

In fact, when you create a Livewire component, it auto-generates 2 files **App/Http/Livewire/{component-class}** and **resources/views/livewire/{component-view}**.

Also, a Livewire component does not rely on controllers to execute a feature, as it performs real time actions in the back and updates (in real time also) the view as per action.

**Comment by undjike** [comment]
The problem is with this default behavior of Livewire (about files auto-generation and auto-discovery).

You know better than I that services in Lucid have their own tree-structure. Surely, if I place a Livewire component in one of my service, Livewire won't detect it.

**Comment by tuannda** [comment]
@Mulkave can you help me implement Livewire with lucid :)

**Reply by undjike** [reply]
Hello @Mulkave, really sorry for the long silence. I was overbooked on a new project. And you know what ... I used Lucid with Livewire 😉.
I'll be happy to submit a pull request if necessary or just explain the procedure to make it work. As soon as possible.

**Reply by tuannda** [reply]
@undjike you can help me

```
just explain the procedure to make it work :)
```

**Reply by undjike** [reply]
First, you need to publish the Livewire's configuration.

Set **class_namespace** to '`App\\Services'`, and **view_path** to `app_path('Services')`

Then run `php artisan livewire:discover` to update components auto-discovery.

You can access your Livewire components like this
`<livewire:{service}.http.{component} />`

**Reply by undjike** [reply]
Let me say
```
<livewire:{path.to.the.component.class} />
```

Including service name.

**Reply by Mulkave** [reply]
@undjike amazing! You've been extremely helpful. I'll be adding those details to the docs a.s.a.p. Thank you!

**Reply by undjike** [reply]
It's a pleasure 😅.

**Reply by dy05** [reply]
Well done.

**Reply by Mulkave** [reply]
@undjike do you mind posting it as an answer instead of a comment? so that we can mark it as the correct answer 😄

**Comment by undjike** [comment, marked as answer]
First, you need to publish the Livewire's configuration.

Set **class_namespace** to `App\\Services`, and **view_path** to `app_path('Services')`

Then run `php artisan livewire:discover` to update components auto-discovery.

You can access your Livewire components like this
`<livewire:{service}.http.{component} />`
